pymc_marketing/mmm/budget_optimizer.py

```diff
@@ -20,6 +21,128 @@
 from pymc_marketing.mmm.transformers import michaelis_menten
 from pymc_marketing.mmm.utils import sigmoid_saturation
 
+from pymc_marketing.mmm.components.adstock import (
+    AdstockTransformation
+)
+from pymc_marketing.mmm.components.saturation import (
+    SaturationTransformation
+)
+
+class BudgetOptimizer:
+    """
+    A class for optimizing budget allocation in a marketing mix model.
```

Comment:
As there are many things we could optimize for, can you please add a description of the logic behind the optimization?